### PR TITLE
Environment restructure

### PR DESCRIPTION
**Pull request by Nishanthh**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,14 +4,16 @@
   "scripts": {
     "ng": "ng",
     "start": "node server.js",
-    "serve": "ng serve --configuration=dev",
-    "start:staging": "gulp set --env=staging && concurrently --kill-others \"node server.js\" \"npm run serve\"",
-    "start:prod": "gulp set --env=prod && concurrently --kill-others \"node server.js\" \"npm run serve\"",
+    "start:staging": "gulp set --env=staging && concurrently --kill-others \"node server.js\" \"npm run serve:staging\"",
+    "start:prod": "gulp set --env=prod && concurrently --kill-others \"node server.js\" \"npm run serve:production\"",
```
Comment (Manzurkds) [inline package.json:8]: @Nishanthh You can replace the content in  `start:` scripts with the one in `serve:` scripts and remove the `serve:` scripts, we won't be using them

Comment (Manzurkds) [reply package.json:8]: @Nishanthh The port value can be given in angular.json file

File: src/app/graphql/graphql.service.ts [hunk 2]
```diff
@@ -24,13 +24,11 @@ export class GraphqlService {
...
-      console.error('Error fetching environment');
-    });
-
-    const localEnvironment = this.environment && this.environment.APP_ENVIRONMENT === 'staging' ? staging : prod;
+    // const env: any = await this.getEnvironment().catch(e => {
+    //   console.error('Error fetching environment');
+    // });
 
-    this.environment = {...localEnvironment, ...this.environment};
+    this.environment = {...environment};
```
Comment (Manzurkds) [inline src/app/graphql/graphql.service.ts:31]: `this.environment = environment`

File: package.json [hunk 3]
```diff
@@ -4,14 +4,14 @@
   "scripts": {
     "ng": "ng",
     "start": "node server.js",
-    "serve": "ng serve --configuration=dev",
-    "start:staging": "gulp set --env=staging && concurrently --kill-others \"node server.js\" \"npm run serve\"",
-    "start:prod": "gulp set --env=prod && concurrently --kill-others \"node server.js\" \"npm run serve\"",
+    "start:staging": "gulp set --env=staging && concurrently --kill-others \"node server.js\" \"ng serve --configuration=dev-staging\"",
+    "start:prod": "gulp set --env=prod && concurrently --kill-others \"node server.js\" \"ng serve --configuration=dev-production\"",
```
Comment (Manzurkds) [inline package.json:8]: @Nishanthh It should have the serve scripts bro
`"start:staging": "ng serve --configuration=dev-staging"

Comment (Manzurkds) [reply package.json:8]: @Nishanthh Also have you given the port value in the angular.json file? I don't see it there

Comment (Nishanthh) [reply package.json:8]: > @Nishanthh It should have the serve scripts bro
> `"start:staging": "ng serve --configuration=dev-staging"

@manzur i have added the scripts.

Comment (Nishanthh) [reply package.json:8]: `        "serve": {
          "builder": "@angular-devkit/build-angular:dev-server",
          "options": {
            "browserTarget": "civis:build",
            "port": 3400
          },
          "configurations": {
            "production": {
              "browserTarget": "civis:build:production"
            },
            "staging": {
              "browserTarget": "civis:build:staging"
            },
            "dev-staging": {
              "browserTarget": "civis:build:dev-staging"
            },
            "dev-production": {
              "browserTarget": "civis:build:dev-production"
            }
          }
        },`
@manzur This is how port added in angular.json file

Comment (Manzurkds) [reply package.json:8]: @Nishanthh I see `"start:staging": "gulp set --env=staging && concurrently --kill-others \"node server.js\" \"ng serve --configuration=dev-staging\"",` whereas it should be 
`"start:staging": "ng serve --configuration=dev-staging"`

**Pull request merged**